geomesa-filter/src/main/scala/org/locationtech/geomesa/filter/FilterHelper.scala

```diff
@@ -418,6 +430,54 @@ object FilterHelper {
     Some(FilterValues(Seq(Bounds.everything[T]), precise = false))
   }
 
+  /**
+   * Calculates all the different case permutations of a string.
+   *
+   * For example, "foo" -> Seq("foo", "Foo", "fOo", "foO", "fOO", "FoO", "FOo", "FOO")
+   *
+   * @param string input string
+   * @return
+   */
+  private def casePermutations(string: String): Seq[String] = {
```

Comment:
Oh man... this is nerd sniping at its best...  It feels like this ought to be something that can be done with the Scala collections library in one line.:)

Reply:
yeah, i had a `foldLeft` that did it, but it built up the arrays one char at a time so i switched to this...

Reply:
i couldn't come up with a more elegant way... but definitely open to suggestions if you've got some. i'm not crazy about this piece of code. and i would have thought there would be a common library somewhere that does this but i didn't find anything with some light googling.

Reply:
Yeah, the FP part of me wants to see a cool foldLeft or some such.  The 'fast' part of me is really, really happy with this.:)